File: 剑指offer/27.二叉树的镜像.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def mirrorTree(self, root: TreeNode) -> TreeNode:
        if not root:
            return None

        root.left = self.mirrorTree(root.left)
        root.right = self.mirrorTree(root.right)

        root.left, root.right = root.right, root.left

        return root

File: 剑指offer/27.二叉树的镜像.cpp
#include <iostream>

struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int x)
        : val(x)
        , left(NULL)
        , right(NULL)
    {
    }
};

class Solution {
public:
    TreeNode* mirrorTree(TreeNode* root)
    {
        if (root == NULL) {
            return root;
        }
        TreeNode* tmp = root->left;
        root->left = mirrorTree(root->right);
        root->right = mirrorTree(tmp);

        return root;
    }
};